geoh5py/ui_json/enforcers.py

```diff
@@ -0,0 +1,350 @@
+        self._errors: list[BaseValidationError] = []
+
+    @classmethod
+    def from_validations(
+        cls, name: str, validations: dict[str, Any], protected: list[str] | None = None
+    ) -> EnforcerPool:
+        """
+        Create enforcers pool from validations.
+
+        :param name: Name of parameter.
+        :param validations: Encodes validations as enforcer type and
+            validation key value pairs.
+        :param protected: Excludes listed enforcer types from updating
+        """
+        enforcers = cls(name)
+        enforcers.update(validations, protected)
+        return enforcers
+
+    def update(self, validations: dict[str, Any], protected: list[str] | None = None):
+        """
+        Create enforcers pool from name/validation dictionary.
+
+        :param validations: Encodes validations as enforcer type and
+            validation key value pairs.
+        :param protected: Excludes listed enforcer types from updating
+
+        """
+        protected = protected or []
+        enforcers = [k for k in self.enforcers if k.enforcer_type in protected]
+        for name, validation in validations.items():
+            if name not in protected:
+                enforcers.append(self._recruit_enforcer(name, validation))
+
+        self.enforcers = enforcers
+
+    @property
+    def validations(self) -> dict[str, Any]:
+        """Returns an enforcer type / validation dictionary from pool."""
+        return {k.enforcer_type: k.validations for k in self.enforcers}
+
+    def _recruit_enforcer(self, enforcer_type: str, validation: Any) -> Enforcer:
+        """
+        Create enforcer from enforcer type and validation.
+
+        :param enforcer_type: Type of enforcer to create.
+        :param validation: Enforcer validation.
+        """
+
+        if enforcer_type not in self.enforcer_types:
+            raise ValueError(f"Invalid enforcer type: {validation['type']}")
+
+        if enforcer_type == "type":
+            enforcer = TypeEnforcer(validation)
+        if enforcer_type == "value":
+            enforcer = ValueEnforcer(validation)  # type: ignore
+        if enforcer_type == "uuid":
+            enforcer = UUIDEnforcer(validation)  # type: ignore
+        if enforcer_type == "required":
+            enforcer = RequiredEnforcer(validation)  # type: ignore
+        if enforcer_type == "required_uijson_parameters":
+            enforcer = RequiredUIJsonParameterEnforcer(validation)  # type: ignore
+        if enforcer_type == "required_form_members":
+            enforcer = RequiredFormMemberEnforcer(validation)  # type: ignore
+        if enforcer_type == "required_workspace_objects":
+            enforcer = RequiredWorkspaceObjectEnforcer(validation)  # type: ignore
+        if enforcer_type == "required_object_data":
+            enforcer = RequiredObjectDataEnforcer(validation)  # type: ignore
+
+        return enforcer
+
+    def validate(self, value: Any):

```

Comment:
Suggest that we keep the same `enforce` method name as the individual enforcers?

Reply:
done